README.md

```diff
@@ -7,210 +7,206 @@
+    - Allows to call Go's methods from Goby directly (only on Linux for now)
+- Builtin multi-threaded server and DB library
+- REPL (run `goby -i`)
 
-### Normal language features
+### Language
+
+Perhaps Goby should be far easier for Rubyists to comprehend. You can use Ruby's syntax highlighting for Goby as well😀
+ 
 - Everything is object
 - Object and Class
     - Top level main object
     - Constructor
-    - Support class methods
-    - Support inheritance
-    - Support instance variable
-    - Support `self`
-- Module & Namespace
-- Variables
-    - Constant
+    - Class/instance method
+    - Class
+        - Inheritance
+        - Singleton class is now supported
+    - `self`
+- Module
+    - `#include`
+    - `#extend`
+    - `::` for namespace
+- Variable: starts with lowercase letter like 'var`
     - Local variable
     - Instance variable
-- Method
-    - Support evaluation with arguments
-    - Support evaluation without arguments
-    - Support evaluation with block (closure)
-- BuiltIn Data Types (All of them are classes 😀)
-    - Class
-    - Integer
-    - String
-    - Boolean
-    - nil
-    - Hash (with built in `to_json` method)
-    - Array
+- Constant
+    - Starts with uppercase like `Var` or `VAR`
+    - global if defined on top-level 
+    - **not reentrant**
+    - (special variables with `$` are unsupported)
+- Methods 
+    - Evaluation with arguments
+    - Evaluation without arguments
+    - Evaluation with block (closure)
+    - Defining singleton method
+- Block
+    - `do` - `end`
 - Flow control
-    - If statement
-    - while statement
-- Import other files
-    - require_relative
-    - require (only for standard libraries now)
-- Standard Libraries (all of them are at very early stage)
-    - `URI`
-    - `Net::HTTP`
-    - `Net::SimpleServer` (This is very cool and quite performante, check the [sample](https://github.com/goby-lang/goby/blob/master/samples/server.gb))
-    - `File`
-    - `DB`
+    - `if`, `else`, `elsif`
+    - `while`
 - IO
-    - `puts`
+    - `#puts`
     - `ARGV`
-- REPL (run `goby -i`)
-- Thread (this should work but the implementation is quite naive and will be refined in the future)
-    - Support `thread` method to create a new thread (like `goroutine`)
-    - Has `Channel` class for passing objects between threads (like `chan` in Go)
+- Import files
+    - `require` (Just for standard libraries by now)
+    - `require_relative`
+- Thread (not a class!)
+    - Goroutine-based `thread` method to create a new thread
+    - Works with `Channel` class for passing objects between threads, like `chan` in Go
     - See this sample: [One thousand threads](https://github.com/goby-lang/goby/blob/master/samples/one_thousand_threads.gb)
 
-    
-**(You can open an issue for any feature request)**
+### Native class
+ 
+Written in Go.
 
-## Install
+- Class
+- Integer
+- String
+- Boolean
+- Null (nil)
+- Hash
+- Array
+- Range
+- URI
+- Channel
+- GoObject (provides `#go_func`)
```

Comment:
GoObject holds pure Go objects(or pointers) and  provides `#go_func` to interact with them